### PR TITLE
separate `projection` and `alias-projection`

### PR DESCRIPTION
**Pull request by jackwener**
# Which issue does this PR close?

Part of #3927 #2212

We should separate `projection` and `alias-projection`.

In fact, `alias-projection` is `subqueryAlias`, after this job, we can easy remove the alias in projection, and replace it with `subqueryAlias` 

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/tests/sql/subqueries.rs [hunk 1]
```diff
@@ -56,13 +56,15 @@ where c_acctbal < (
...
-                Projection: lineitem.l_orderkey, SUM(lineitem.l_extendedprice) AS price AS __value, alias=__sq_1
-                  Aggregate: groupBy=[[lineitem.l_orderkey]], aggr=[[SUM(lineitem.l_extendedprice)]]
-                    TableScan: lineitem projection=[l_orderkey, l_extendedprice]"#
+          Projection: orders.o_custkey
+            Aggregate: groupBy=[[orders.o_custkey]], aggr=[[SUM(orders.o_totalprice)]]
+              Filter: CAST(orders.o_totalprice AS Decimal128(25, 2)) < __sq_1.__value
+                Inner Join: orders.o_orderkey = __sq_1.l_orderkey
+                  TableScan: orders projection=[o_orderkey, o_custkey, o_totalprice]
+                  Projection: lineitem.l_orderkey, SUM(lineitem.l_extendedprice) AS price AS __value, alias=__sq_1
+                    Projection: lineitem.l_orderkey
```
Comment (jackwener) [inline datafusion/core/tests/sql/subqueries.rs:65]: Problems like this indicate `pushdown` rule exist some bug.

**Comment by mingmwang** [comment]
Do we need to support such kind of usage?
````
select ('aa1', 'aa2' as AA1) as B
````

**Comment by mingmwang** [comment]
BTW, below is the Spark SQL's plan

````
== Optimized Logical Plan ==
Project [[aa1,aa2] AS B#941145], Statistics(sizeInBytes=6.0 B)
+- OneRowRelation, Statistics(sizeInBytes=1.0 B)

== Physical Plan ==
*(1) Project [[aa1,aa2] AS B#941145]
+- *(1) Scan OneRowRelation[]

````

**Comment by jackwener** [comment]
Spark doesn't have alias-projection. @mingmwang 

We also can read `sql standard 2003` 

Alias just contain 
- `expr-alias (like: projection contain is expr)` 
- `table-ref-alias`

```
select ('aa1', 'aa2' as AA1) as B

is Alias("B", Expr::List(aa1, Alias("AA1", aa2)))
```

Spark projection just contains `Seq[NamedExpr]` `input`

**Comment by jackwener** [comment]
https://imgse.com/i/znjM9I

**Comment by jackwener** [comment]
> BTW, below is the Spark SQL's plan
> 
> ```
> == Optimized Logical Plan ==
> Project [[aa1,aa2] AS B#941145], Statistics(sizeInBytes=6.0 B)
> +- OneRowRelation, Statistics(sizeInBytes=1.0 B)
> 
> == Physical Plan ==
> *(1) Project [[aa1,aa2] AS B#941145]
> +- *(1) Scan OneRowRelation[]
> ```
> 
> Another example;
> 
> `select ('aa1', 'aa2' as AA1) as B, ('xx1', 'xx2' as BB1) as X`
> 
> I think we should keep the alias-projection, and alias-projection can perfect represent such usage patten.

These alias bo

**Comment by mingmwang** [comment]
> https://imgse.com/i/znjM9I
> 
> So we don't need projection keep alias, just expr::Alias is enough

I am OK with removing alias from Projection. Let get some advices from other members also.

**Comment by jackwener** [comment]
close. all job finish in #4333

**Pull request closed without merging**